### PR TITLE
Usage field support in OpenAI API - unary

### PR DESCRIPTION
**Pull request by mzegla**
### 🛠 Summary

Addressing: https://github.com/openvinotoolkit/model_server/issues/2626

- Adding `usage` field to completions and chat completions endpoints
- Removing old, obsolete calculator

### 🧪 Checklist

- [x] Unit tests added.
- [ ] The documentation updated. // Will update when streaming part is also added
- [x] Change follows security best practices.
``

### CODE REVIEW

File: src/llm/http_llm_calculator.cc [hunk 1]
```diff
@@ -51,6 +51,12 @@ enum class Endpoint {
     COMPLETIONS,
 };
 
+struct CompletionUsageStatistics {
+    size_t promptTokens = 0;
+    size_t completionTokens = 0;
+    size_t totalTokens = 0;
```
Comment (dkalinowski) [inline src/llm/http_llm_calculator.cc:57]: this field is unused, remove?

Comment (dkalinowski) [reply src/llm/http_llm_calculator.cc:57]: i would suggest adding method "getTotalTokens" that simply adds prompt tokens and completion tokens. this way we could re-use this code in serialization in multiple places

File: src/test/llmnode_test.cpp [hunk 2]
```diff
@@ -145,6 +145,12 @@ TEST_F(LLMFlowHttpTest, unaryCompletionsJson) {
     }
     ASSERT_EQ(d["model"], "llmDummyKFS");
     ASSERT_EQ(d["object"], "text_completion");
+
+    ASSERT_TRUE(d["usage"].IsObject());
+    ASSERT_TRUE(d["usage"].GetObject()["prompt_tokens"].IsInt());
+    ASSERT_TRUE(d["usage"].GetObject()["completion_tokens"].IsInt());
+    ASSERT_TRUE(d["usage"].GetObject()["total_tokens"].IsInt());
+    ASSERT_EQ(d["usage"].GetObject()["completion_tokens"].GetInt(), 5 /* max_tokens */);
```
Comment (dkalinowski) [inline src/test/llmnode_test.cpp:153]: can we really assert the number? the model output can change

File: src/llm/http_llm_calculator.cc (hunk 1 above)
Comment (mzegla) [reply src/llm/http_llm_calculator.cc:57]: I created it to make this structure in-line with OpenAI `usage` field spec, but sure I can apply your suggestion.

File: src/test/llmnode_test.cpp (hunk 2 above)
Comment (mzegla) [reply src/test/llmnode_test.cpp:153]: I think it's pretty safe. We set max_tokens to very low number and it's very unlikely that model will naturally stop generating after less than 5 tokens.

**Review by rasapala — APPROVED** [review]

File: src/llm/http_llm_calculator.cc [hunk 3]
```diff
@@ -51,6 +51,15 @@ enum class Endpoint {
     COMPLETIONS,
 };
 
+struct CompletionUsageStatistics {
+    size_t promptTokens = 0;
+    size_t completionTokens = 0;
+
+    size_t calculateTotalTokens() {
```
Comment (dkalinowski) [inline src/llm/http_llm_calculator.cc:58]:
```suggestion
    size_t calculateTotalTokens() const {
```
can be const

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**